### DISCUSSION TITLE
Sensor Status: Sensor Failed 7

### DISCUSSION THREAD

**Discussion by pkacp1** [NightscoutFoundation/xDrip#3600] [Bug reports]
![Screenshot_20240727-194314](https://github.com/user-attachments/assets/8a8e1d24-5ba1-4857-86bc-60b9c18c44f3)
Since switching to Dexcom One+, I have been experiencing issues with sensor connection. After 5-7 days, the sensor stops connecting with xDrip. In System Status, I receive a Sensor Status: Sensor Failed 7 or SensorError message.

Normally, I would assume the sensor had failed, but if I disconnect it from xDrip, wait 15 minutes, and then connect it to the original Dexcom App on another phone, it works well until the end of its declared lifetime.

Does anyone have any ideas or has anyone experienced the same problem? I found few raport on similar issue, but in all cases the sensor wasn’t working on Dexcom One+ app anymore so clearly the sensor failed. This is not my case.

**Comment by Navid200** [comment]
If there is a connectivity issue, I can see switching to the official app could solve the problem.
The recommended settings as well as how to deal with intermittent connectivity is included here:
https://navid200.github.io/xDrip/docs/Intermittent.html


But, if the problem is a sensor error, I would be surprised if switching to the official app would solve the problem.

**Comment by pkacp1** [comment]
I don't think the is with the connection with transmiter/sensor itself, because when I look on System Status screen every 5 minutes it get;s connected, but don't get readings. I have no idea how this works, but is looks to me like from time to time the data that is sent by the sensor is not understandable by xDrip - does it make any sense? What "Sensor Fail 7" command means?

**Comment by pstojak** [comment]
Hi,
having the same issue as pkacp1, just on new sensor. We're using One+ for couple on months now.
Sensor Fail 7 + seems to be connected but xDrip can't understand data. 
On dexcom one+ Receiver everything is ok. 
Seems like xDrip having issues with decoding 'new' sensor data format?

**Reply by Navid200** [reply]
There is no way to proceed without knowing what the phone brand and model is and see what the status page looks like.
There is nothing wrong with xDrip as many are using it.  

You can have one setting incorrectly and nothing will work.  For example, not having disabled xDrip battery optimization or not having disabled Bluetooth battery optimization.  
And those are only 2 examples.

**Reply by pstojak** [reply]
Thanks for quick answer. We reinstalled everything on the mobile phone and it's ok now.

**Comment by jamorham** [comment]
I have made a modification in the latest nightly to treat this error type 7 as to allowing glucose values to be retrieved so we should see if this happens again and see if that change helps.

**Comment by pkacp1** [comment]
I have it again after 5 days, but this time the sensor isn't working in One+ App either. This is the log I get in xDrip:
![Screenshot (6 wrz 2024 23_09_01)](https://github.com/user-attachments/assets/4cdba54d-909f-407f-a006-67301057ee9e)

**Reply by VitaliySh** [reply]
I have the same issue with my Honor X9b. The logs have been uploaded.
![photo_2024-10-09_09-28-53](https://github.com/user-attachments/assets/55289694-0983-4db7-976f-7f21cd4478d2)

**Reply by Navid200** [reply]
@VitaliySh This thread was created a long time ago.  xDrip has been updated a lot since then.
The option for uploading the logs is there for a purpose.  It's not for everyone to upload.  There is only one person who can see your uploaded logs.  There are others who can provide support here.  

In general, if a sensor does not work with Dexcom One+ app, it is most likely not going to work with xDrip either.
With the new xDrip release, if you see sensor failed 7 and get no readings, it means that something is wrong with the sensor.  But, you should always try Dexcom app to confirm.
